docs/docs/tutorials/deploying_and_interacting_with_smart_contracts.md

```diff
@@ -0,0 +1,382 @@
+
+// Get the ABI from the compiled contract
+const abi = compiledCode.contracts[fileName][contractName].abi;
+
+// Write the Contract ABI to a new file
+const abiPath = path.join(__dirname, 'MyContractAbi.json');
+fs.writeFileSync(abiPath, JSON.stringify(abi, null, '\t'));
+
+// Log the Contract ABI to the console
+console.log('Contract ABI:\n', abi);
+```
+
+This code reads the Solidity code from the `MyContract.sol` file, compiles it using `solc`, and generates the ABI and bytecode for the smart contract. It then writes the bytecode to a new file called `MyContractBytecode.bin` and the contract ABI to `MyContractAbi.json`. And it logs them to the console.
+
+Run the following command to compile the Solidity code:
+
+```
+node compile.js
+```
+
+If everything is working correctly, you should see both the Contract Bytecode and the Contract ABI logged to the console.
+
+:::tip
+📝 There are couple of other ways to get the bytecode and the ABI like using Truffle development framework and running `truffle compile` (https://trufflesuite.com/docs/truffle/quickstart/#compile).
+Another way is to use Remix and check the _Compilation Details_ after compiling the smart contract (https://remix-ide.readthedocs.io/en/latest/run.html#using-the-abi-with-ataddress)
+:::
+
+## Step 5: Set up web3.js and connect to the Ganache network
+
+In this step, we will set up the web3.js library and connect to the Ganache network. So, be sure to run Ganache if you did not already did.
+
+First, install the `web3` package using npm:
+
+```
+npm install web3@4.0.1-rc.1
+```
+
+Note that we are installing the latest version of 4.x, at the time of this tutorial writing. You can check the latest version at https://www.npmjs.com/package/web3?activeTab=versions
+
+Next, create a new file called `index.js` in your project directory and add the following code to it:
+
+```javascript
+const { Web3 } = require('web3');
+
+// Set up a connection to the Ganache network
+const web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:7545'));
+
+// Log the current block number to the console
+web3.eth
+	.getBlockNumber()
+	.then(result => {
+		console.log('Current block number: ' + result);
+	})
+	.catch(error => {
+		console.error(error);
+	});
+```
+
+This code sets up a connection to the Ganache network and logs the current block number to the console.
+
+Run the following command to test the connection:
+
+```
+node index.js
+```
+
+If everything is working correctly, you should see the current block number logged to the console. However, if you got an error with the reason `connect ECONNREFUSED 127.0.0.1:7545` then double check that you are running Ganache locally on port `7545`.
+
+## Step 5: Deploy the smart contract to the Ganache network using web3.js
+
+In this step, we will use web3.js to deploy the smart contract to the Ganache network.
+
+Create a file named `deploy.js` and fill it with the following code:
+
+```javascript
+const { Web3 } = require('web3');
```

Comment:
wt do you think about encouraging users to use individual packages instead of importing main web3, as it brings many not-required packages as well , if some one is not doing optimizations for that non-required pkgs.

Reply:
My 2c: tutorias are meant for people first time interacting with web3.js. Goal is to make it as simple as possible. Using individual packages would make sense in some "Advanced tutorials"

Reply:
@jdevcs @mpetrunic 
Thanks, and yes, I kept it using one package for simplicity. But I should add a hint regarding this.

@avkos 
Thanks for sharing this interesting information. Do we have it mentioned somewhere? If not, I think we should highlight this in multiple places.

Reply:
@avkos auto signing works if there is [wallet in context](https://github.com/web3/web3.js/blob/ae1d156e4b6cf17ab641b1e96cebf9cacd2a1e41/packages/web3-eth/src/rpc_method_wrappers.ts#L565). 

for example one of working snippit:
```
const {Wallet, privateKeyToAccount} = require('web3-eth-accounts');
const {Contract} = require('web3-eth-contract');
const {Web3Context} = require('web3-core');

const con_json = require('contract.json');

async function test() {
    
//create a context with wallet and provider
    const account = privateKeyToAccount("PRIVATE KEY");
    const context = new Web3Context(
        { 
            provider: 'http://127.0.0.1:7545',
            wallet: new Wallet(account)
        });

//pass that contract to other packages instead of only provider
    const contract = new Contract(con_json.abi, context);
    
    const contractInstance = await contract.deploy(
        {
            data: con_json.bytecode
        }).send({ 
            gas: 10000,
            from: account.address });

    console.log(contractInstance.options.address);


    const receipt = await contractInstance.methods.store(12).send(
        {
            from: account.address, 
            gas: 10000
        });

    console.log(receipt);


    //.....
}


```

However as @mpetrunic said , it should not be added in current tutorial , so we will not do it.

Reply:
Thanks for example @jdevcs , It will be great to have this example somewhere in tutorials